mrp_lot_number_propagation/models/mrp_production.py

```diff
@@ -0,0 +1,174 @@
+                continue
+            order.is_lot_number_propagated = propagate_lot
+            propagate_move = order.move_raw_ids.filtered(
+                lambda m: m.bom_line_id.propagate_lot_number
+            )
+            if not propagate_move:
+                raise UserError(
+                    _(
+                        "Bill of material is marked for lot number propagation, but "
+                        "there are no components propagating lot number. "
+                        "Please check BOM configuration."
+                    )
+                )
+            elif len(propagate_move) > 1:
+                raise UserError(
+                    _(
+                        "Bill of material is marked for lot number propagation, but "
+                        "there are multiple components propagating lot number. "
+                        "Please check BOM configuration."
+                    )
+                )
+            else:
+                propagate_move.propagate_lot_number = True
+
+    def _post_inventory(self, cancel_backorder=False):
+        self._create_and_assign_propagated_lot_number()
+        return super()._post_inventory(cancel_backorder=cancel_backorder)
+
+    def _create_and_assign_propagated_lot_number(self):
+        for order in self:
+            if not order.is_lot_number_propagated or order.lot_producing_id:
+                continue
+            finish_moves = order.move_finished_ids.filtered(
+                lambda mv, mo=order: mv.product_id == mo.product_id
+                and mv.state not in ("done", "cancel")
+            )
+            if finish_moves and not finish_moves.quantity_done:
+                lot_model = self.env["stock.lot"]
+                lot = lot_model.search(
+                    [
+                        ("product_id", "=", order.product_id.id),
+                        ("company_id", "=", order.company_id.id),
+                        ("name", "=", order.propagated_lot_producing),
+                    ],
+                    limit=1,
+                )
+                if lot.quant_ids:
+                    raise UserError(
+                        _(
+                            "Lot/Serial number %s already exists and has been used. "
+                            "Unable to propagate it."
+                        )
+                    )
+                if not lot:
+                    lot = self.env["stock.lot"].create(
+                        {
+                            "product_id": order.product_id.id,
+                            "company_id": order.company_id.id,
+                            "name": order.propagated_lot_producing,
+                        }
+                    )
+                order.with_context(lot_propagation=True).lot_producing_id = lot
+
+    def write(self, vals):
+        for order in self:
+            if (
+                order.is_lot_number_propagated
+                and "lot_producing_id" in vals
```

Comment:
@astirpe if lot_producing_id is False we should not raise the error

Reply:
@CasVissers-360ERP done here https://github.com/OCA/manufacture/pull/1299/commits/c62f997d23aaf9b98f80a10058f22c88a6317458 could you check?

Reply:
Thnx resolved.